api/src/main/java/io/kafbat/ui/service/KafkaConnectService.java

```diff
@@ -39,23 +43,67 @@
 
 @Service
 @Slf4j
-@RequiredArgsConstructor
 public class KafkaConnectService {
   private final ClusterMapper clusterMapper;
   private final KafkaConnectMapper kafkaConnectMapper;
   private final KafkaConfigSanitizer kafkaConfigSanitizer;
+  private final ClustersProperties clustersProperties;
+
+  private final AsyncCache<ConnectCacheKey, List<InternalConnectorInfo>> cachedConnectors;
+
+  public KafkaConnectService(ClusterMapper clusterMapper, KafkaConnectMapper kafkaConnectMapper,
+                             KafkaConfigSanitizer kafkaConfigSanitizer,
+                             ClustersProperties clustersProperties) {
+    this.clusterMapper = clusterMapper;
+    this.kafkaConnectMapper = kafkaConnectMapper;
+    this.kafkaConfigSanitizer = kafkaConfigSanitizer;
+    this.clustersProperties = clustersProperties;
+    this.cachedConnectors = Caffeine.newBuilder()
+        .expireAfterWrite(clustersProperties.getCache().getConnectCacheExpiry())
```

Comment:
you can
```
 expireAfterWrite(
   clustersProperties.getCache().isEnabled() 
     ? clustersProperties.getCache().getConnectCacheExpiry()
     : Duration.ZERO
 )
 ```
 and just call it without further isEnabled checks

Reply:
It's look a bit unclear in this case as for me